### DISCUSSION TITLE
EJBCA Rest Client enabled: still showing this service has been disabled

### DISCUSSION THREAD

**Discussion by JibinNajeeb** [Keyfactor/ejbca-ce#199] [General]
Hi, I have followed the docker instruction to run EJBCA-CE. 
1) Executed below command
`docker run -it --rm -p 80:8080 -p 443:8443 -h mycahostname primekey/ejbca-ce`
2) Enable Rest-api as shown in below screen
![image](https://user-images.githubusercontent.com/47626138/209150653-fad93fb9-b9d5-4e7f-9ad6-328884e2dd49.png)
3) Tried a sample API
![image](https://user-images.githubusercontent.com/47626138/209150729-093f0222-851e-4b4e-889f-722d3b10765a.png)

It is showing service is disabled. 
Could anyone please help, what is it I'm doing wrong?
Thanks

**Comment by eryabies** [comment]
Hi!

Same issue with the latest ejbca-ce container.

**Comment by primetomas** [comment]
@JibinNajeeb : Looking at #201, it looks like you got it working.

Although I could reproduce it on the container.

**Comment by primetomas** [comment]
Ok, it has to do with trying to use the REST API with the browser. In the log file you can see:

Custom header for browser calls is absent with forbidden header: HttpServletRequestImpl [ GET /ejbca/ejbca-rest-api/v1/certificate/CN%3DManagement%20CA%2CO%3DPK%2CC%3DSE/1234567890ABCDEF/revocationstatus

Using CURL it works fine, example command:
curl -k --cert-type P12 --cert superadmin.p12:password -X GET "https://mycahostname/ejbca/ejbca-rest-api/v1/certificate/CN%3DManagement%20CA%2CO%3DPK%2CC%3DSE/1234567890ABCDEF/revocationstatus" -H  "accept: application/json"

Response:
{"error_code":404,"error_message":"CA 'CN=Management CA,O=PK-DM,C=AE' does not exist."}

You can find a setting for this below the protocol configuration, as "Security measures for REST endpoint", uncheck "Mandate custom header for REST endpoint" and it works from the browser as well.

**Comment by JibinNajeeb** [comment]
I am getting below error. Tested in 7.9.0.2
Management CA exists but getting below error
![image](https://user-images.githubusercontent.com/47626138/211793624-3054d8fc-0609-4301-ab76-bd13ed8f223d.png)

**Reply by primetomas** [reply]
That is because your Management CA does not have the subject DN "CN=Management CA,O=PK-DM,C=SE". You need to edit the curl command so the subject DN matches the subject DN of your managment CA. It's URL encoded so "O%3DPK" is the "O=PK", you can simply change "PK" so "Sample CA" or whatever the O field is in your Management CA Subject DN.

The REST call itself seems to work fine :-)
